src/main/java/com/google/sps/servlets/PersonServlet.java

```diff
@@ -49,5 +50,19 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             request.getParameter("company"),
             request.getParameter("job"),
             request.getParameter("linkedin")));
+    // TODO: redirect to home page and handle in JS.
+  }
+
+  // Returns the person the request's email belongs to.
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Optional<Person> personOpt = personDao.get(request.getParameter("email"));
```

Comment:
This seems fishy. You're allowing any user to query for any other user by email, and sending them back the user's full name, company, job, and linked in address?

This data is technically PII (personally identifiable information), and probably should not just be returned to whomever asks for it.

It looks like you're intending to only have the user themselves fetch this. Can we more deeply integrate the authentication system to ensure that requester is *authenticated as* the user they're looking up?

Reply:
l can add a check to ensure that.

Reply:
SGTM, thanks.

Reply:
Also depends on Person class, which needs to be refactored to Autovalue style. Another PR.